components/bind/Solaris/dns-server.sh

```diff
@@ -99,40 +104,97 @@ case "$method" in
 	    set -- `echo ${value} | /usr/bin/sed -e  's/\\\\//g'`
 	    server=$@
 	    ;;
+	'user')
+	    cmdopts="${cmdopts} -u ${value}"
+	    cmduser=${value};
+	    ;;
 	esac
     done
 
+    configuration_dir=$(sed -n -e 's,^[[:space:]]*directory.*"\(.*\)";,\1,p' \
+        ${configuration_file})
+    [ "${configuration_dir}" == "" ] && configuration_dir=/etc/namedb
+
+    configuration_files=$(sed -n -e \
+        "s,^[[:space:]]*file.*\"\(.*\)\";,${configuration_dir}/\1,p" \
+        ${configuration_file} | sort -u)
+    configuration_files="${configuration_files} ${configuration_file}"    
+}
+
+result=${SMF_EXIT_OK}
+
+# Read command line arguments
+method="$1"		# %m
+instance="$2" 		# %i
+contract="$3"		# %{restarter/contract}
+
+# Set defaults; SMF_FMRI should have been set, but just in case.
+if [ -z "$SMF_FMRI" ]; then
+    SMF_FMRI="svc:/network/dns/server:${instance}"
+fi
+server="/usr/sbin/named"
+I=`/usr/bin/basename $0`
+
+case "$method" in
+'start')
+    get_config
+
+    # Check configuration file exists.
+    if [ ! -f ${configuration_file} ]; then
+        msg="$I : Configuration file ${configuration_file} does not exist!"
+        echo ${msg} >&2
+        /usr/bin/logger -p daemon.error ${msg}
+        # dns-server should be placed in maintenance state.
+        result=${SMF_EXIT_ERR_CONFIG}
+    fi
+
     # If chroot option is set, note zones(5) are preferred, then
     # configuration file lives under chroot directory.
     if [ "${chroot_dir}" != "" ]; then
-      configuration_file=${chroot_dir}${configuration_file}
-      rndc_config_file=${chroot_dir}${rndc_config_file}
-      rndc_key_file=${chroot_dir}${rndc_key_file}
-      rndc_cmd_opts="${rndc_cmd_opts} -t ${chroot_dir}"
+        if [ "${chroot_dir}" = "/" ]; then
+            msg="$I : chroot_dir must not be /"
+            echo ${msg} >&2
+            /usr/bin/logger -p daemon.error ${msg}
+            # dns-server should be placed in maintenance state.
+            exit ${SMF_EXIT_ERR_CONFIG}
+        fi
+
+        server="env LD_NOLAZYLOAD=1 ${server}"
+
+        mkdir -p ${chroot_dir}
```

Comment:
It's not an error for mkdir -p if a directory already exists, it will leave it alone.